tests/baselines/reference/commonMissingSemicolons.errors.txt

```diff
@@ -253,4 +261,42 @@ tests/cases/compiler/commonMissingSemicolons.ts(51,1): error TS2304: Cannot find
     varMyVar;
     ~~~~~~~~
 !!! error TS2304: Cannot find name 'varMyVar'.
-    
\ No newline at end of file
+    
+    class NoSemicolonClassA {
+      ['a'] = 0
+      {}
+      ~
+!!! error TS1005: ';' expected.

```

Comment:
Ok, so, in this case, we don’t need a semicolon because we had a line break and `{` is not one of the tokens that prevents ASI from happening. So here, we do want `Unexpected token. A constructor, method, accessor, or property was expected.` But if it were on the previous line, we’d want `';' expected.` Sorry if I led you astray in my previous comment—I think what I said was right, but incomplete. This stuff is tricky 😅 Some test cases:


```ts
class A {
  foo = 0 {} // ';' expected
}

class B {
  foo = 0; {} // Unexpected token. A constructor, method, accessor, or property was expected.
}

class C {
  foo = 0
  {} // Unexpected token. A constructor, method, accessor, or property was expected.
}
```

which is what they all say in current master.

Reply:
Oh thanks, this helps clear it up. I think I'll put a little more effort into getting good specific error messages for this case after all.

Reply:
I went with the existing error message after all; the longer one gave me pause (it felt like a tl;dr).